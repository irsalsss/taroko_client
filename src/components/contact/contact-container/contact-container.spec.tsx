import {
  render,
  screen,
  userEvent,
  waitFor,
} from "@/utils/test/react-testing-setup";
import { wrapperReactQuery } from "@/utils/test/wrapper-testing";
import ContactContainer from "./contact-container";

describe("ContactContainer", () => {
  it("should render ContactContainer successfully", () => {
    const { baseElement } = render(<ContactContainer />, {
      wrapper: wrapperReactQuery,
    });

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  it("should delete contact", async () => {
    render(<ContactContainer />, {
      wrapper: wrapperReactQuery,
    });

    await waitFor(() => {
      const deleteIcon = screen.getAllByRole("button", {
        name: "delete-icon",
      })[0];
      expect(deleteIcon).toBeVisible();

      userEvent.click(deleteIcon);
    });

    await waitFor(() => {
      const modalTitle = screen.getByText("Delete Contact");
      expect(modalTitle).toBeVisible();
    });

    const submitBtn = screen.getByRole("button", { name: /submit/i });
    userEvent.click(submitBtn);

    await waitFor(() => {
      const modalTitle = screen.queryByText("Delete Contact");
      expect(modalTitle).toBeNull();
    });
  });

  it("should favorite contact", async () => {
    render(<ContactContainer />, {
      wrapper: wrapperReactQuery,
    });

    await waitFor(() => {
      const favoriteIcon = screen.getAllByRole("button", {
        name: "star-icon",
      })[0];
      expect(favoriteIcon).toBeVisible();

      userEvent.click(favoriteIcon);
    });

    await waitFor(() => {
      const favoriteIcon = screen.getAllByRole("button", {
        name: "star-filled-icon",
      })[0];
      expect(favoriteIcon).toBeVisible();
    });
  });

  it("should sort by descending", async () => {
    render(<ContactContainer />, {
      wrapper: wrapperReactQuery,
    });

    await waitFor(() => {
      const ascendingIcon = screen.getByRole("button", {
        name: "asc-icon",
      });
      expect(ascendingIcon).toBeVisible();

      userEvent.click(ascendingIcon);
    });

    await waitFor(() => {
      const descendingIcon = screen.getByRole("button", {
        name: "desc-icon",
      });
      expect(descendingIcon).toBeVisible();
    });
  });
});
