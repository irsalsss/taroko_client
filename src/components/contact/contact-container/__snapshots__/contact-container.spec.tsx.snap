// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactContainer should render ContactContainer successfully 1`] = `
<body>
  <div>
    <div
      class="p-4"
    >
      <div
        class="flex flex-wrap items-center justify-between w-full px-4 gap-4"
      >
        <h3>
          Contact List
        </h3>
        <div
          class="relative flex gap-1 flex-col"
        >
          <div
            class="input-icon"
          >
            <svg
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10 6.5C10 8.433 8.433 10 6.5 10C4.567 10 3 8.433 3 6.5C3 4.567 4.567 3 6.5 3C8.433 3 10 4.567 10 6.5ZM9.30884 10.0159C8.53901 10.6318 7.56251 11 6.5 11C4.01472 11 2 8.98528 2 6.5C2 4.01472 4.01472 2 6.5 2C8.98528 2 11 4.01472 11 6.5C11 7.56251 10.6318 8.53901 10.0159 9.30884L12.8536 12.1464C13.0488 12.3417 13.0488 12.6583 12.8536 12.8536C12.6583 13.0488 12.3417 13.0488 12.1464 12.8536L9.30884 10.0159Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <input
            class="input input-padding-with-icon"
            placeholder="search anything here..."
            type="text"
            value=""
          />
        </div>
        <button
          class="button"
          label="Add"
          role="button"
        >
          Add
        </button>
      </div>
      <div
        class="flex items-center justify-between px-4 mt-4"
      >
        <div
          class="flex items-center gap-3 w-full"
        >
          <div
            class="tab tab-active"
          >
            <span>
              All
            </span>
          </div>
          <div
            class="tab"
          >
            <span>
              Favorites
            </span>
          </div>
        </div>
        <button
          class="button-icon"
          role="button"
        >
          <span
            class="sr-only"
          >
            asc-icon
          </span>
          <svg
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M3.89949 9.49998C3.89949 9.72089 3.7204 9.89997 3.49949 9.89997C3.27857 9.89997 3.09949 9.72089 3.09949 9.49998L3.09949 2.46566L1.78233 3.78282C1.62612 3.93903 1.37285 3.93903 1.21664 3.78282C1.06043 3.62661 1.06043 3.37334 1.21664 3.21713L3.21664 1.21713C3.29166 1.14212 3.3934 1.09998 3.49949 1.09998C3.60557 1.09998 3.70732 1.14212 3.78233 1.21713L5.78233 3.21713C5.93854 3.37334 5.93854 3.62661 5.78233 3.78282C5.62612 3.93903 5.37285 3.93903 5.21664 3.78282L3.89949 2.46566L3.89949 9.49998ZM8.49998 1.99998C8.22383 1.99998 7.99998 2.22383 7.99998 2.49998C7.99998 2.77612 8.22383 2.99998 8.49998 2.99998H14.5C14.7761 2.99998 15 2.77612 15 2.49998C15 2.22383 14.7761 1.99998 14.5 1.99998H8.49998ZM8.49998 4.99998C8.22383 4.99998 7.99998 5.22383 7.99998 5.49998C7.99998 5.77612 8.22383 5.99998 8.49998 5.99998H14.5C14.7761 5.99998 15 5.77612 15 5.49998C15 5.22383 14.7761 4.99998 14.5 4.99998H8.49998ZM7.99998 8.49998C7.99998 8.22383 8.22383 7.99998 8.49998 7.99998H14.5C14.7761 7.99998 15 8.22383 15 8.49998C15 8.77612 14.7761 8.99998 14.5 8.99998H8.49998C8.22383 8.99998 7.99998 8.77612 7.99998 8.49998Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <div
        class="flex justify-center items-center p-8 mt-8"
      >
        <h5>
          The 
          contacts
           
           are empty...
        </h5>
      </div>
      <div
        class="flex flex-wrap gap-4 px-4 mt-4"
      />
    </div>
  </div>
</body>
`;
